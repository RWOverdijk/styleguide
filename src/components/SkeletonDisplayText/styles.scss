/** @define SkeletonDisplayText */

@import "../../css/_global";
@import "../../css/mixins";

.Van-SkeletonDisplayText {
  @include skeleton-shimmer;
  @include skeleton-style;

  backface-visibility: hidden;
  max-width: 12rem;
  position: relative;

  &:not(:first-child) {
    margin-top: 1.2rem;
  }

  &:not(:last-child) {
    margin-bottom: 1.2rem;
  }

  &--small {
    height: 2.8rem;
  }

  &--medium {
    height: 3.2rem;
  }

  &--large {
    height: 4.4rem;
  }

  &--extraLarge {
    height: 6.4rem;
  }

  &--full {
    max-width: 100%;
    width: 100%;
  }
}
