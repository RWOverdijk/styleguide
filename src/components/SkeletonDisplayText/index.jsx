import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import './styles.scss';

class SkeletonDisplayText extends PureComponent {
  render() {
    const {
      className,
      size,
      full,
      ...props
    } = this.props;
    return (
      <div
        className={classnames('Van-SkeletonDisplayText', {
          'Van-SkeletonDisplayText--small': size === 'small',
          'Van-SkeletonDisplayText--medium': size === 'medium',
          'Van-SkeletonDisplayText--large': size === 'large',
          'Van-SkeletonDisplayText--extraLarge': size === 'extraLarge',
          'Van-SkeletonDisplayText--full': full,
        }, className)}
        {...props}
      />
    );
  }
}

SkeletonDisplayText.defaultProps = {
  className: '',
  size: 'medium',
  full: false,
};

SkeletonDisplayText.propTypes = {
  className: PropTypes.string,
  size: PropTypes.oneOf([
    'small',
    'medium',
    'large',
    'extraLarge',
  ]),
  full: PropTypes.bool,
};

export default SkeletonDisplayText;
