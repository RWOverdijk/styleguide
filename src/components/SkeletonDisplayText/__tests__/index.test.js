import React from 'react';
import renderer from 'react-test-renderer';
import SkeletonDisplayText from '../';

describe("<SkeletonCard>", () => {
  it("renders without params", () => {
    const tree = renderer.create(
      <SkeletonDisplayText />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders small size", () => {
    const tree = renderer.create(
      <SkeletonDisplayText size="small" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders medium size", () => {
    const tree = renderer.create(
      <SkeletonDisplayText size="medium" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders large size", () => {
    const tree = renderer.create(
      <SkeletonDisplayText size="large" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders extraLarge size", () => {
    const tree = renderer.create(
      <SkeletonDisplayText size="extraLarge" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders full width", () => {
    const tree = renderer.create(
      <SkeletonDisplayText full />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders small size and full width", () => {
    const tree = renderer.create(
      <SkeletonDisplayText size="small" full />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific className", () => {
    const tree = renderer.create(
      <SkeletonDisplayText className="my-class" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("renders with specific attributes", () => {
    const tree = renderer.create(
      <SkeletonDisplayText data-id="0" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
