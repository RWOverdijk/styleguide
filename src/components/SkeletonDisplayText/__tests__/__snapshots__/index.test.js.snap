// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SkeletonCard> renders extraLarge size 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--extraLarge"
/>
`;

exports[`<SkeletonCard> renders full width 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--medium Van-SkeletonDisplayText--full"
/>
`;

exports[`<SkeletonCard> renders large size 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--large"
/>
`;

exports[`<SkeletonCard> renders medium size 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--medium"
/>
`;

exports[`<SkeletonCard> renders small size 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--small"
/>
`;

exports[`<SkeletonCard> renders small size and full width 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--small Van-SkeletonDisplayText--full"
/>
`;

exports[`<SkeletonCard> renders with specific attributes 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--medium"
  data-id="0"
/>
`;

exports[`<SkeletonCard> renders with specific className 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--medium my-class"
/>
`;

exports[`<SkeletonCard> renders without params 1`] = `
<div
  className="Van-SkeletonDisplayText Van-SkeletonDisplayText--medium"
/>
`;
